lib/process-manager.ts

```diff
@@ -57,40 +60,88 @@ class RawSubprocessStream extends Streams.ObjectReadWriteStream<string> {
 		this.process = process;
 	}
 	_write(message: string) {
-		if (!this.process.getProcess().connected) {
+		if (!this.process.isConnected) {
 			// no error because the crash handler should already have shown an error, and
 			// sometimes harmless messages are sent during cleanup
 			return;
 		}
-		this.process.process.send(message);
+		this.process._send(message);
 		// responses are handled in ProcessWrapper
 	}
 }
 
-interface ProcessWrapper {
-	load: number;
-	process: ChildProcess | Worker;
-	release: () => Promise<void>;
-	getProcess: () => ChildProcess;
+abstract class ProcessWrapper {
+	process: ChildProcess | null = null;
+	clusterWorker: cluster.Worker | null = null;
+	worker: worker_threads.Worker | null = null;
+	isConnected = true;
+	pid = 0;
+	constructor(file: string, type: ProcessManager['type'] = 'process', env?: AnyObject) {
+		switch (type) {
+		case 'process':
+			this.process = child_process.fork(file, [], {cwd: ROOT_DIR, env});
+			this.process.on('disconnect', () => {
+				this.isConnected = false;
+			});
+			this.pid = this.process.pid;
+			break;
+		case 'cluster':
+			this.clusterWorker = cluster.fork(env);
+			this.process = this.clusterWorker.process;
+			this.clusterWorker.on('disconnect', () => {
+				this.isConnected = false;
+			});
+			this.pid = this.process.pid;
+			break;
+		case 'worker':
+			this.worker = new Worker(file, {env});
```

Comment:
```suggestion
			this.worker = new Worker(file, {workerData: env});
```
While i was transitioning #7350 to ProcessWrapper, i noticed in testing that this crashes unless changed to this.

Reply:
I think this is just JavaScript being extra dumb.

I'd do this, probably:

```suggestion
			const options = {};
			if (env) options.env = env;
			this.worker = new Worker(file, options);
```

Reply:
This also crashes. It explicitly needs env to be `{workerData: env}` as i suggested, or else workerData will be undefined in the worker.

Reply:
Wait, that makes no sense. The documentation doesn't say anything about requiring `workerData`...

Reply:
Exactly how and why does it crash? The current code works fine for me in `verifier`. Is it Windows that's crashing, or is it the way you're using it?

Reply:
Are you using it by passing non-JSON to env? Is that why it's crashing?

Reply:
I don't use Windows. What i'm doing is passing the data i would set as options.workerData property directly - the error i get is that workerData is undefined in the worker.
Code: 
```ts
	constructor(options: SQLOptions) {
		super(__filename, 'worker', options);
```

Reply:
...that's not what it's for.

Reply:
Of course if you put workerData in the env field, it won't appear as workerData. Did you realize that's what you were doing?

Reply:
I see. My bad, then.